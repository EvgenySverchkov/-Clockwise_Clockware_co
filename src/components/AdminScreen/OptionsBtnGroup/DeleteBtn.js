import React from 'react';

export default function DeleteBtn({deleteMasterById, id}){
  return (
    <button onClick={()=>deleteMasterById(id)} className="dropdown-item">
      Delete
    </button>
  );
}
