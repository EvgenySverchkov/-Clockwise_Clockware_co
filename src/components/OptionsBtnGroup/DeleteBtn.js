import React from "react";
import PropTypes from "prop-types";

function DeleteBtn({ deleteMasterById, id }) {
  return (
    <button onClick={() => deleteMasterById(id)} className="dropdown-item">
      Delete
    </button>
  );
}

DeleteBtn.propTypes = {
  deleteMasterById: PropTypes.func,
  id: PropTypes.number,
};

export default DeleteBtn;
