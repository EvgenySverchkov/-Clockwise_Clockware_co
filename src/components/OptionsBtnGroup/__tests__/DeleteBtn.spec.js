import React from "react";
import {create} from "react-test-renderer";

import DeleteBtn from "../DeleteBtn";

describe("Test of <DeleteBtn />", ()=>{
    const mockProps = {
        deleteMasterById: jest.fn(), 
        id: 1
    }
    it("Default view", ()=>{
        const component = create(<DeleteBtn {...mockProps}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("Check call of handler", ()=>{
        const component = create(<DeleteBtn {...mockProps}/>);
        component.root.findByType("button").props.onClick()
        expect(mockProps.deleteMasterById).toHaveBeenCalledTimes(1);
    });
});