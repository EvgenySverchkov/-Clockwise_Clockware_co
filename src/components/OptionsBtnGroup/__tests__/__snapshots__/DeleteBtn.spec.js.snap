// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <DeleteBtn /> Default view 1`] = `
<button
  className="dropdown-item deleteItemBtn"
  onClick={[Function]}
>
  Delete
</button>
`;
